### PR TITLE
feat(theme): component style && common theme

### PR DESCRIPTION
**Pull request by wenmine**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Release Notes

- **New Features**
	- Introduced a more flexible theme management system, allowing dynamic selection of themes based on user settings.
	- Added comprehensive CSS variable definitions for various components, enhancing customization and consistency across the application.

- **Bug Fixes**
	- Improved scrollbar customization and text overflow handling for better visual presentation.

- **Removals**
	- Removed several outdated theme-related files and classes, streamlining the codebase and focusing on essential components.

- **Updates**
	- Updated CSS variable references to align with a restructured design system.
	- Modified CSS styles across various components to improve visual consistency and presentation.
	- Enhanced visual representation of various components through updated color and style definitions.
	- Adjusted margins, paddings, and background colors across multiple components for improved layout and spacing.

### CODE REVIEW

**Comment by hexqi** [comment]
image
页面JS UI有问题

**Review by hexqi — COMMENTED** [review]
image
Tab样式不太对

File: packages/theme/base/src/page/base-config-page.less [hunk 1]
```diff
@@ -3,12 +3,36 @@ html {
 }
 
 body {
-  padding: 0;
-  margin: 0;
-  font-family: Microsoft YaHei, Microsoft YaHei-Normal;
-}
+  *,
```
Comment (gene9831) [inline packages/theme/base/src/page/base-config-page.less:6]: `*` 是通用选择器，下面的 `.lowcode-scrollbar` 和 `.lowcode-scrollbar-thin` 是否有必要？

Comment (wenmine) [reply packages/theme/base/src/page/base-config-page.less:6]: 通用选择器比class选择器的权重低很多，这边还是保留

File: packages/common/component/MetaCodeEditor.vue [hunk 2]
```diff
@@ -328,7 +328,7 @@ export default {
     color: var(--ti-lowcode-meta-code-editor-header-tips-container-color);
 
     .header-tips-title {
-      color: var(--ti-lowcode-base-text-color-3);
+      color: var();
```
Comment (hexqi) [inline packages/common/component/MetaCodeEditor.vue]: 这里是不是可以去掉

**Comment by hexqi** [comment]
image
国际化插件UI不对

**Comment by wenmine** [comment]
> image 国际化插件UI不对

image

File: packages/theme/base/src/component-common.less [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+  .tiny-form-item__error {
+    font-size: var(--te-base-font-size);
+  }
+}
+
+// 19、表格
+.tiny-grid__wrapper {
+  .tiny-grid {
+    font-size: var(--te-base-font-size-base);
+    background-color: var(--te-common-text-primary);
```
Comment (hexqi) [inline packages/theme/base/src/component-common.less]: 背景色不应该是文本色，表格背景会变成黑色

**Review by hexqi — DISMISSED** [review]
LGTM

File: packages/theme/base/src/common.less [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  --te-common-bg-primary-emphasize: var(--te-base-blue-6); // 主色-背景色 #1476ff
+  --te-common-bg-secondary: var(--te-base-gray-65); // 次要按钮-背景色 #B4B5B8
+  --te-common-bg-prompt: var(--te-base-gray-66); // 导航按钮-背景色/选块-选中色 #8B8C8F
+  --te-common-bg-container: var(
+    --te-base-gray-68
+  ); // 列表悬浮色/表格悬浮色/顶部导航按钮悬浮色/选块背景色/提示背景色 #36383D
+  --te-common-bg-container-weaken: var(--te-base-gray-69); // 浅编辑代码--背景色 #2B2D33
+  --te-common-bg-tag: var(--te-base-dark-28); // tag标签-背景色 #1d312a
+  --te-common-bg-default: var(--te-base-dark-27); // 深色背景-输入框背景/面板背景色 #212329
+  --te-common-bg-component: var(--te-base-gray-1); // 深色背景-输入框等组件默认背景/面板背景色 #212329
```
Comment (hexqi) [inline packages/theme/base/src/common.less]: te-base-gray-1 是#fff，和注释不太匹配

File: packages/theme/base/src/component-common.less [hunk 5]
```diff
@@ -0,0 +1,759 @@
...
+    font-size: var(--te-base-font-size-base);
+    > a {
+      color: var(--te-common-text-link);
+    }
+  }
+  .tiny-notify__message-zone {
+    margin: 0 0 0 8px;
+    .tiny-notify__title-wrapper {
+      font-size: var(--te-base-font-size-1);
+      font-weight: var(--te-base-font-weight-4t);
```
[the remark this answers is not in the document]
Comment (hexqi) [reply packages/theme/base/src/component-common.less]: 多了一个t

File: packages/theme/base/src/component-common.less [hunk 6]
```diff
@@ -0,0 +1,759 @@
...
+    .tiny-numeric__increase {
+      width: 14px;
+      border-left: 1px solid var(--te-common-border-default);
+    }
+  }
+  .tiny-numeric-base .is-disabled {
+    fill: var(---te-common-bg-disabled);
+    background: var(---te-common-bg-disabled);
+    color: var(--te-common-text-disabled);
+  }
```
[the remark this answers is not in the document]
Comment (hexqi) [reply packages/theme/base/src/component-common.less]: 多了一个-

**Comment by wenmine** [comment]
image
 zhe'g'y这个样式很丑，需要确认

**Review by hexqi — DISMISSED** [review]

File: packages/settings/styles/src/js/useStyle.js [hunk 7]
```diff
@@ -177,19 +177,21 @@ watch(
...
-
-    // 解析css
-    const { parseList, selectors, styleObject } = parser(value)
-
-    state.cssParseList = parseList
-    state.selectors = selectors
-    state.styleObject = styleObject
-  }
-)
+setTimeout(() => {
```
Comment (chilingling) [inline packages/settings/styles/src/js/useStyle.js:180]: 这里不要用延时，用事件通知

File: packages/theme/base/src/page/base-config-page.less [hunk 8]
```diff
@@ -129,3 +119,15 @@ body {
     display: none;
   }
 }
+
+.global-desc-info {
+  color: #808080;
```
Comment (chilingling) [inline packages/theme/base/src/page/base-config-page.less]: 这里为什么不用变量？

File: packages/theme/base/src/component-common.less
Comment (chilingling) [inline packages/theme/base/src/component-common.less:1]: 这个 `component-common`  文件是不是对 tinyvue 组件的定制化？
后续是需要保留的吗？如果需要保留，是不是每个组件拆开，这样可读性好一点？

File: packages/theme/base/src/index.js [hunk 9]
```diff
@@ -1,15 +1,26 @@
...
+import './component-common.less'
 
-export const tinyEngineThemeLight = {
-  id: 'tiny-engine-light-theme',
-  name: 'EngineLightTheme',
-  cnName: 'EngineLightTheme',
-  data: { ...tinyBaseTheme, ...concatThemeLight }
+export const tinyThemeLightVars = {
+  id: 'tiny-light-theme',
+  name: 'tinyLightTheme',
```
Comment (chilingling) [inline packages/theme/base/src/index.js]: `tiny-xxx-theme` 改成 `tiny-theme-xxx` 这样感觉顺一点？

File: packages/theme/base/src/index.js [hunk 10]
```diff
@@ -1,15 +1,26 @@
...
+  cnName: '',
```
Comment (chilingling) [inline packages/theme/base/src/index.js:12]: 这样的国际化是不是标准一点？

```javascript
{
  name: {
     zh_CN:  '',
    en_US:  'tiny-xxx'
  }
}
```

File: packages/settings/styles/src/js/useStyle.js (hunk 7 above)
Comment (wenmine) [reply packages/settings/styles/src/js/useStyle.js:180]: 现在的事件通知机制是啥样的？有示例吗？

File: packages/theme/base/src/index.js (hunk 10 above)
Comment (wenmine) [reply packages/theme/base/src/index.js:12]: image
TinyVue的格式要求

File: packages/settings/styles/src/js/useStyle.js (hunk 7 above)
Comment (wenmine) [reply packages/settings/styles/src/js/useStyle.js:180]: 之前事件通知还没好，这里几个遗留，到时候统一梳理整改吧

File: packages/theme/base/src/component-common.less
Comment (wenmine) [reply packages/theme/base/src/component-common.less:1]: 就是对 tinyvue 组件的定制化，我感觉写在一个文件里就行，集中一点，调整也就是在一个文件里统一调整了

File: packages/theme/base/src/base.less [hunk 11]
```diff
@@ -0,0 +1,514 @@
...
+  --te-base-dark-default-button-border-hover-color: #e6e6e6;
+
+  // 分割线
+  --te-base-split-color-1: rgba(0, 0, 0, 0.08); // 白色背景分割线
+  --te-base-split-color-2: rgba(0, 0, 0, 0.15); // 灰色背景分割线
+  --te-base-dark-split-color-1: rgba(255, 255, 255, 0.08); // 白色背景分割线
+  --te-base-dark-split-color-2: rgba(255, 255, 255, 0.15);
+
+  // rgba
+  --te-base-rgba-1: rgba(0, 0, 0, 0.2);
```
Comment (chilingling) [inline packages/theme/base/src/base.less:408]: 这堆 rgba 哪里来的？有具体的意义吗？

File: packages/theme/base/src/base.less [hunk 12]
```diff
@@ -478,7 +478,7 @@
   /**
    * 1.3 字号
   **/
-  --te-base-font-size: 11px; // 正文-常规
+  --te-base-font-size-sm: 11px; // 正文-常规
```
Comment (chilingling) [inline packages/theme/base/src/base.less]: 后面的注释也跟着更新一下吧

File: packages/theme/base/src/base.less (hunk 11 above)
Comment (wenmine) [reply packages/theme/base/src/base.less:408]: 目前用作在box-shadow里，到时候让别歌重新梳理一下

File: packages/theme/base/src/page/base-config-page.less (hunk 8 above)
Comment (wenmine) [reply packages/theme/base/src/page/base-config-page.less]: 改啦

**Review by chilingling — APPROVED** [review]

**Review by hexqi — APPROVED** [review]

**Pull request merged**